src/main/java/seedu/address/model/activity/Activity.java

```diff
@@ -124,8 +182,71 @@ public void addExpense(Expense ... expenditures) throws PersonNotInActivityExcep
         if (!allPresent) {
             throw new PersonNotInActivityException();
         }
-        for (int i = 0; i < expenditures.length; i++) {
-            expenses.add(expenditures[i]);
+        for (Expense expense : expenditures) {
+            expenses.add(expense);
+
+            // We update the balance sheet
+            int payer = expense.getPersonId();
+            double amount = expense.getAmount().getValue();
+            double splitAmount = amount / participantIds.size();
+            int i = 0;
+            for (; i < participantIds.size(); i++) {
+                if (participantIds.get(i) == payer) {
+                    break;
+                }
+            }
+
+            for (int j = 0; j < debtMatrix.size(); j++) {
+                if (j != i) {
+                    debtMatrix.get(j).set(i, debtMatrix.get(j).get(i) + splitAmount);
+                }
+            }
+        }
+    }
+
+    /**
+     * Simplifies the expenses in the balance sheet and also updates transferMatrix.
+     * See: https://pure.tue.nl/ws/portalfiles/portal/2062204/623903.pdf
+     */
+    private void simplifyExpenses() {
+        int i = 0;
+        int j = 0;
+        int n = participantBalances.size();
+
+        // negative balance means you lent more than you borrowed.
+        for (int a = 0; a < debtMatrix.size(); a++) {
+            double acc = 0;
+            for (int b = 0; b < debtMatrix.size(); b++) {
+                acc += debtMatrix.get(a).get(b);
+                acc -= debtMatrix.get(b).get(a);
+                transferMatrix.get(a).set(b, 0.0);
+            }
+            participantBalances.set(a, acc);
+        }
+
+        System.out.println("Balances: " + participantBalances.toString());
```

Comment:
Is this `println` statement intended as logging output?

Reply:
oh no there's no way to expose it now so this is the only way to check